osc-server/src/main/java/org/osc/core/broker/service/validator/DeleteDistributedApplianceRequestValidator.java

```diff
@@ -62,4 +71,43 @@ public DistributedAppliance validateAndLoad(BaseDeleteRequest request) throws Ex
 
         return da;
     }
+
+    private void validateDAReferenceToSfcAndSecurityGroup(EntityManager em, DistributedAppliance da) throws Exception {
+		for (VirtualSystem vs : da.getVirtualSystems()) {
+			validateVSReferenceToSfcAndSecurityGroup(em, da, vs);
+		}
+	}
+
+	private void validateVSReferenceToSfcAndSecurityGroup(EntityManager em, DistributedAppliance da, VirtualSystem vs)
+			throws Exception {
+
+		if (!getSFCs(vs)) {
```

Comment:
This is not improving the readability. It is also not reducing nesting. Instead simply start this method with if vs.getsfcs == null || vs.getsfcs.isempty then return immediately.